src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCharClass.cs

```diff
@@ -517,8 +517,7 @@ public void AddSubtraction(RegexCharClass sub)
         public void AddRange(char first, char last)
         {
             _rangelist.Add(new SingleRange(first, last));
-            if (_canonical && _rangelist.Count > 0 &&
-                first <= _rangelist[_rangelist.Count - 1].Last)
+            if (_canonical && first <= last)
```

Comment:
I suppose this could logically omit the check that canonical is true, at the expense of extra "writes"

Reply:
Yes.